soc/arm/nordic_nrf/nrf91/soc.c

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief System/hardware module for Nordic Semiconductor nRF91 family processor
+ *
+ * This module provides routines to initialize and support board-level hardware
+ * for the Nordic Semiconductor nRF91 family processor.
+ */
+
+#include <kernel.h>
+#include <init.h>
+#include <cortex_m/exc.h>
+#include <nrfx.h>
+#include <soc/nrfx_coredep.h>
+
+#ifdef CONFIG_RUNTIME_NMI
+extern void _NmiInit(void);
+#define NMI_INIT() _NmiInit()
+#else
+#define NMI_INIT()
+#endif
+
+#if defined(CONFIG_SOC_NRF9160)
+#include <system_nrf9160.h>
+#else
+#error "Unknown SoC."
+#endif
+
+#include <nrf.h>
+#include <hal/nrf_power.h>
```

Comment:
These two inclusions do not seem to be needed (`<nrf.h>` is included by `<nrfx.h>`, and POWER HAL is not used here).

Reply:
OK, removed.
Note that the inclusion of nrf.h is present in the nrf51,51 soc.c files, too, so we need to clean this up from those, too.